// @rmwc/slider/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Slider = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var base_1 = require("@rmwc/base");
var foundation_1 = require("./foundation");
var SliderTrack = react_1.default.memo(react_1.default.forwardRef(function SliderTrack(props, ref) {
    return react_1.default.createElement("div", { ref: ref, className: "mdc-slider__track" });
}));
var SliderTrackMarkerContainer = react_1.default.memo(react_1.default.forwardRef(function SliderTrackMarkerContainer(props, ref) {
    return react_1.default.createElement("div", { ref: ref, className: "mdc-slider__track-marker-container" });
}));
var SliderPin = react_1.default.memo(function SliderPin(_a) {
    var value = _a.value;
    return (react_1.default.createElement("div", { className: "mdc-slider__pin" },
        react_1.default.createElement("span", { className: "mdc-slider__pin-value-marker" }, value)));
});
var SliderThumb = react_1.default.memo(function SliderThumb() {
    return (react_1.default.createElement("svg", { className: "mdc-slider__thumb", width: "21", height: "21" },
        react_1.default.createElement("circle", { cx: "10.5", cy: "10.5", r: "7.875" })));
});
var SliderFocusRing = react_1.default.memo(function SliderFocusRing() {
    return react_1.default.createElement("div", { className: "mdc-slider__focus-ring" });
});
exports.Slider = base_1.createComponent(function Slider(props, ref) {
    var _a = foundation_1.useSliderFoundation(props), rootEl = _a.rootEl, thumbContainerEl = _a.thumbContainerEl, sliderPinEl = _a.sliderPinEl, setTrackRef = _a.setTrackRef, setTrackMarkerContainerRef = _a.setTrackMarkerContainerRef;
    var value = props.value, min = props.min, max = props.max, discrete = props.discrete, displayMarkers = props.displayMarkers, step = props.step, disabled = props.disabled, onChange = props.onChange, onInput = props.onInput, children = props.children, foundationRef = props.foundationRef, rest = tslib_1.__rest(props, ["value", "min", "max", "discrete", "displayMarkers", "step", "disabled", "onChange", "onInput", "children", "foundationRef"]);
    var className = base_1.useClassNames(props, [
        'mdc-slider',
        {
            'mdc-slider--discrete': discrete,
            'mdc-slider--display-markers': displayMarkers && discrete
        }
    ]);
    var dataStep = step ? { 'data-step': step } : {};
    if (displayMarkers && !discrete) {
        console.warn("The 'displayMarkers' prop on rmwc Slider will\n        only work in conjunction with the 'discrete' prop");
    }
    return (react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tabIndex: 0, 
        //eslint-disable-next-line jsx-a11y/role-has-required-aria-props
        role: "slider", "aria-valuemax": max, "aria-valuenow": value, "aria-label": "Select Value" }, (disabled ? { 'aria-disabled': disabled } : {}), dataStep, rest, { ref: ref, element: rootEl, className: className }),
        react_1.default.createElement("div", { className: "mdc-slider__track-container" },
            react_1.default.createElement(SliderTrack, { ref: setTrackRef }),
            displayMarkers && (react_1.default.createElement(SliderTrackMarkerContainer, { ref: setTrackMarkerContainerRef }))),
        react_1.default.createElement(base_1.Tag, { element: thumbContainerEl, className: "mdc-slider__thumb-container" },
            discrete && react_1.default.createElement(SliderPin, { value: sliderPinEl.getProp('value') }),
            react_1.default.createElement(SliderThumb, null),
            react_1.default.createElement(SliderFocusRing, null)),
        children));
});
