// @rmwc/menu/dist/menu/foundation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useMenuFoundation = void 0;
var tslib_1 = require("tslib");
var react_1 = require("react");
var menu_1 = require("@material/menu");
var base_1 = require("@rmwc/base");
exports.useMenuFoundation = function (props) {
    var menuSurfaceApi = react_1.useRef();
    var listApi = react_1.useRef();
    var setListApi = function (api) {
        listApi.current = api;
    };
    var setMenuSurfaceApi = function (api) {
        menuSurfaceApi.current = api;
    };
    var items = react_1.useCallback(function () {
        var _a;
        return ((_a = listApi.current) === null || _a === void 0 ? void 0 : _a.listElements()) || [];
    }, []);
    var _a = base_1.useFoundation({
        props: tslib_1.__assign(tslib_1.__assign({}, props), { 
            // we don't want to pass the apiRef all the way through
            apiRef: undefined }),
        elements: {
            rootEl: true
        },
        foundation: function (_a) {
            var emit = _a.emit;
            return new menu_1.MDCMenuFoundation({
                addClassToElementAtIndex: function () {
                    var _a;
                    var args = [];
                    for (var _i = 0; _i < arguments.length; _i++) {
                        args[_i] = arguments[_i];
                    }
                    return (_a = listApi.current) === null || _a === void 0 ? void 0 : _a.addClassToElementIndex.apply(_a, tslib_1.__spread(args));
                },
                removeClassFromElementAtIndex: function () {
                    var _a;
                    var args = [];
                    for (var _i = 0; _i < arguments.length; _i++) {
                        args[_i] = arguments[_i];
                    }
                    return (_a = listApi.current) === null || _a === void 0 ? void 0 : _a.removeClassFromElementAtIndex.apply(_a, tslib_1.__spread(args));
                },
                addAttributeToElementAtIndex: function () {
                    var _a;
                    var args = [];
                    for (var _i = 0; _i < arguments.length; _i++) {
                        args[_i] = arguments[_i];
                    }
                    return (_a = listApi.current) === null || _a === void 0 ? void 0 : _a.setAttributeForElementIndex.apply(_a, tslib_1.__spread(args));
                },
                removeAttributeFromElementAtIndex: function (index, attr) {
                    var list = items();
                    list[index].removeAttribute(attr);
                },
                elementContainsClass: function (element, className) {
                    return element.classList.contains(className);
                },
                closeSurface: function () {
                    var _a;
                    (_a = menuSurfaceApi.current) === null || _a === void 0 ? void 0 : _a.setOpen(false);
                },
                getElementIndex: function (element) {
                    return items().indexOf(element);
                },
                notifySelected: function (evtData) {
                    return emit('onSelect', {
                        index: evtData.index,
                        item: items()[evtData.index]
                    });
                },
                getMenuItemCount: function () { var _a; return ((_a = listApi.current) === null || _a === void 0 ? void 0 : _a.getListItemCount()) || 0; },
                focusItemAtIndex: function () {
                    var _a;
                    var args = [];
                    for (var _i = 0; _i < arguments.length; _i++) {
                        args[_i] = arguments[_i];
                    }
                    return (_a = listApi.current) === null || _a === void 0 ? void 0 : _a.focusItemAtIndex.apply(_a, tslib_1.__spread(args));
                },
                focusListRoot: function () { var _a; return (_a = listApi.current) === null || _a === void 0 ? void 0 : _a.focusRoot(); }
            });
        }
    }), foundation = _a.foundation, elements = tslib_1.__rest(_a, ["foundation"]);
    var rootEl = elements.rootEl;
    var handleClick = react_1.useCallback(function (evt) {
        var _a;
        (_a = props.onClick) === null || _a === void 0 ? void 0 : _a.call(props, evt);
        // fixes an issue with nested span element on list items
        var el = base_1.closest(evt.target, '.mdc-list-item');
        el && foundation.handleItemAction(el);
    }, [foundation, props.onClick]);
    var handleKeydown = react_1.useCallback(function (evt) {
        var _a;
        (_a = props.onKeyDown) === null || _a === void 0 ? void 0 : _a.call(props, evt);
        foundation.handleKeydown(evt);
        // Jump through some hoops to find out
        // that we are selecting the list item
        // This is instead of trying to listen to an event on the list item
        // which is what MDC does
        if (evt.which === 13 &&
            evt.target instanceof Element &&
            listApi.current &&
            evt.target.classList.contains(listApi.current.getClasses())) {
            foundation.handleItemAction(evt.target);
        }
    }, [foundation, props.onKeyDown]);
    var handleOpen = react_1.useCallback(function (evt) {
        var _a;
        var list = items();
        if ((props.focusOnOpen || props.focusOnOpen === undefined) &&
            list.length > 0 &&
            !list.some(function (el) { return el === document.activeElement; })) {
            list[0].focus();
        }
        (_a = props.onOpen) === null || _a === void 0 ? void 0 : _a.call(props, evt);
    }, [props.onOpen, props.focusOnOpen, items]);
    rootEl.setProp('onKeyDown', handleKeydown, true);
    rootEl.setProp('onClick', handleClick, true);
    rootEl.setProp('onOpen', handleOpen, true);
    var canSetApi = listApi.current && menuSurfaceApi.current && props.apiRef;
    react_1.useEffect(function () {
        if (listApi.current && menuSurfaceApi.current && props.apiRef) {
            props.apiRef(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, listApi.current), menuSurfaceApi.current), { items: items }));
        }
        // eslint-disable-next-line
    }, [canSetApi, items]);
    return tslib_1.__assign({ setListApi: setListApi, setMenuSurfaceApi: setMenuSurfaceApi }, elements);
};
