// @rmwc/menu/dist/menu/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleMenuSurface = exports.SimpleMenu = exports.Menu = exports.MenuItem = exports.MenuItems = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var list_1 = require("@rmwc/list");
var base_1 = require("@rmwc/base");
var menu_surface_1 = require("../menu-surface");
var foundation_1 = require("./foundation");
/** A wrapper for menu items */
exports.MenuItems = base_1.createComponent(function MenuItems(props, ref) {
    var className = base_1.useClassNames(props, ['mdc-list mdc-menu__items']);
    return react_1.default.createElement(list_1.List, tslib_1.__assign({ role: "menu" }, props, { className: className, ref: ref }));
});
exports.MenuItems.displayName = 'MenuItems';
/** This is just the ListItem component exported from the Menu module for convenience. You can use `ListItem` or `SimpleListItem` components from the List section as long as you add `role="menuitem"` and `tabIndex="0"` to the components for accessibility. */
exports.MenuItem = base_1.createComponent(function MenuItem(props, ref) {
    return react_1.default.createElement(list_1.ListItem, tslib_1.__assign({ role: "menuitem", tabIndex: 0 }, props, { ref: ref }));
});
var isMenuItems = function (child) {
    return base_1.getDisplayName(child) === 'MenuItems';
};
/** A menu component for displaying lists items. */
exports.Menu = base_1.createComponent(function Menu(props, ref) {
    var children = props.children, focusOnOpen = props.focusOnOpen, onSelect = props.onSelect, foundationRef = props.foundationRef, rest = tslib_1.__rest(props, ["children", "focusOnOpen", "onSelect", "foundationRef"]);
    var _a = foundation_1.useMenuFoundation(props), rootEl = _a.rootEl, setListApi = _a.setListApi, setMenuSurfaceApi = _a.setMenuSurfaceApi;
    var needsMenuItemsWrapper = (react_1.default.Children.map(children, isMenuItems) || []).every(function (val) { return val === false; });
    var menuItemsProps = {
        apiRef: setListApi
    };
    return (react_1.default.createElement(menu_surface_1.MenuSurface, tslib_1.__assign({}, rootEl.props(rest), { "aria-hidden": !rest.open, className: base_1.classNames('mdc-menu', rest.className), apiRef: setMenuSurfaceApi, ref: ref }), needsMenuItemsWrapper ? (react_1.default.createElement(exports.MenuItems, tslib_1.__assign({}, menuItemsProps), children)) : (react_1.default.Children.map(children, function (child) {
        if (isMenuItems(child)) {
            return react_1.default.cloneElement(child, tslib_1.__assign(tslib_1.__assign({}, (react_1.default.isValidElement(child) ? child.props : {})), menuItemsProps));
        }
        return child;
    }))));
});
var simpleMenuFactory = function (MenuComponent) {
    return function (props) {
        var _a = tslib_1.__read(react_1.useState(!!props.open), 2), stateOpen = _a[0], setStateOpen = _a[1];
        react_1.useEffect(function () {
            if (props.open !== undefined && props.open !== stateOpen) {
                setStateOpen(!!props.open);
            }
        }, [props.open, stateOpen]);
        var handle = props.handle, onClose = props.onClose, children = props.children, _b = props.rootProps, rootProps = _b === void 0 ? {} : _b, open = props.open, foundationRef = props.foundationRef, rest = tslib_1.__rest(props, ["handle", "onClose", "children", "rootProps", "open", "foundationRef"]);
        var wrappedHandle = react_1.default.cloneElement(handle, tslib_1.__assign(tslib_1.__assign({}, handle.props), { onClick: function (evt) {
                setStateOpen(!stateOpen);
                if (handle.props.onClick) {
                    handle.props.onClick(evt);
                }
            } }));
        var wrappedOnClose = function (evt) {
            setStateOpen(!!open || false);
            onClose === null || onClose === void 0 ? void 0 : onClose(evt);
        };
        var RenderMenuComponent = MenuComponent;
        return (react_1.default.createElement(menu_surface_1.MenuSurfaceAnchor, tslib_1.__assign({}, rootProps),
            react_1.default.createElement(RenderMenuComponent, tslib_1.__assign({}, rest, { onClose: wrappedOnClose, open: stateOpen }), children),
            wrappedHandle));
    };
};
/** A Simplified menu component that allows you to pass a handle element and will automatically control the open state and add a MenuSurfaceAnchor */
exports.SimpleMenu = simpleMenuFactory(exports.Menu);
/** The same as SimpleMenu, but a generic surface. */
exports.SimpleMenuSurface = simpleMenuFactory(menu_surface_1.MenuSurface);
