// @rmwc/data-table/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleDataTable = exports.DataTableCell = exports.DataTableHeadCell = exports.DataTableRow = exports.DataTableBody = exports.DataTableHead = exports.DataTableContent = exports.DataTable = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var icon_1 = require("@rmwc/icon");
var base_1 = require("@rmwc/base");
/** The DataTable Component. */
exports.DataTable = base_1.createComponent(function DataTable(props, ref) {
    var stickyColumns = props.stickyColumns, stickyRows = props.stickyRows, rest = tslib_1.__rest(props, ["stickyColumns", "stickyRows"]);
    var className = base_1.useClassNames(props, [
        'mdc-data-table',
        {
            'rmwc-data-table--sticky-columns': !!stickyColumns,
            'rmwc-data-table--sticky-columns-1': !!stickyColumns,
            'rmwc-data-table--sticky-rows': !!stickyRows,
            'rmwc-data-table--sticky-rows-1': !!stickyRows
        }
    ]);
    return (react_1.default.createElement(base_1.DataTableContext.Provider, { value: true },
        react_1.default.createElement(base_1.Tag, tslib_1.__assign({}, rest, { ref: ref, className: className }))));
});
/** The data table content. */
exports.DataTableContent = base_1.createComponent(function DataTableContent(props, ref) {
    var className = base_1.useClassNames(props, ['mdc-data-table__table']);
    return react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "table" }, props, { ref: ref, className: className }));
});
/** A header for the data table. */
exports.DataTableHead = base_1.createComponent(function DataTableHead(props, ref) {
    var className = base_1.useClassNames(props, ['rmwc-data-table__head']);
    return (react_1.default.createElement(base_1.DataTableHeadContext.Provider, { value: true },
        react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "thead" }, props, { ref: ref, className: className }))));
});
/** A body for the data table. */
exports.DataTableBody = base_1.createComponent(function DataTableBody(props, ref) {
    var className = base_1.useClassNames(props, ['mdc-data-table__content']);
    return react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "tbody" }, props, { ref: ref, className: className }));
});
/** A row for the data table. */
exports.DataTableRow = base_1.createComponent(function DataTableRow(props, ref) {
    var isHeaderRow = react_1.useContext(base_1.DataTableHeadContext);
    var activated = props.activated, selected = props.selected, rest = tslib_1.__rest(props, ["activated", "selected"]);
    var className = base_1.useClassNames(props, [
        'rmwc-data-table__row',
        {
            'mdc-data-table__header-row': isHeaderRow,
            'mdc-data-table__row': !isHeaderRow,
            'mdc-data-table__row--selected': props.selected || props.activated,
            'rmwc-data-table__row--activated': props.activated
        }
    ]);
    return react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "tr" }, rest, { ref: ref, className: className }));
});
var DataTableSortIcon = react_1.default.memo(function DataTableSortIcon() {
    return (react_1.default.createElement(icon_1.Icon, { className: "rmwc-data-table__sort-icon", icon: react_1.default.createElement("svg", { width: "18", height: "18", viewBox: "0 0 24 24" },
            react_1.default.createElement("path", { fill: "currentColor", d: "M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z" })) }));
});
/** A header cell for the data table. */
exports.DataTableHeadCell = base_1.createComponent(function DataTableHeadCell(props, ref) {
    var alignStart = props.alignStart, alignMiddle = props.alignMiddle, alignEnd = props.alignEnd, isNumeric = props.isNumeric, sort = props.sort, onSortChange = props.onSortChange, onClick = props.onClick, children = props.children, hasFormControl = props.hasFormControl, rest = tslib_1.__rest(props, ["alignStart", "alignMiddle", "alignEnd", "isNumeric", "sort", "onSortChange", "onClick", "children", "hasFormControl"]);
    var className = base_1.useClassNames(props, [
        'rmwc-data-table__cell',
        'mdc-data-table__header-cell',
        {
            'rmwc-data-table__head-cell--sortable': sort !== undefined,
            'rmwc-data-table__head-cell--sorted': !!sort,
            'rmwc-data-table__head-cell--sorted-ascending': sort === 1,
            'rmwc-data-table__head-cell--sorted-descending': sort === -1,
            'mdc-data-table__header-cell--checkbox': hasFormControl,
            'mdc-data-table__header-cell--numeric': isNumeric,
            'rmwc-data-table__cell--align-start': alignStart,
            'rmwc-data-table__cell--align-middle': alignMiddle,
            'rmwc-data-table__cell--align-end': alignEnd
        }
    ]);
    var onClickProp = sort !== undefined
        ? {
            onClick: function (evt) {
                onSortChange &&
                    onSortChange(sort === null ? 1 : sort === 1 ? -1 : null);
                onClick && onClick(evt);
            }
        }
        : {};
    return (react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "th" }, rest, onClickProp, { className: className, ref: ref, role: "columnheader", scope: "col" }),
        sort !== undefined && react_1.default.createElement(DataTableSortIcon, null),
        children));
});
/** A cell for the DataTable */
exports.DataTableCell = base_1.createComponent(function DataTableCell(props, ref) {
    var alignStart = props.alignStart, alignMiddle = props.alignMiddle, alignEnd = props.alignEnd, isNumeric = props.isNumeric, hasFormControl = props.hasFormControl, rest = tslib_1.__rest(props, ["alignStart", "alignMiddle", "alignEnd", "isNumeric", "hasFormControl"]);
    var className = base_1.useClassNames(props, [
        'mdc-data-table__cell',
        'rmwc-data-table__cell',
        {
            'mdc-data-table__cell--numeric': isNumeric,
            'mdc-data-table__cell--checkbox': hasFormControl,
            'rmwc-data-table__cell--align-start': props.alignStart,
            'rmwc-data-table__cell--align-middle': props.alignMiddle,
            'rmwc-data-table__cell--align-end': props.alignEnd
        }
    ]);
    return react_1.default.createElement(base_1.Tag, tslib_1.__assign({ tag: "td" }, rest, { ref: ref, className: className }));
});
/** A simple data table to render matrices. */
function SimpleDataTable(props) {
    var data = props.data, headers = props.headers, _a = props.getRowProps, getRowProps = _a === void 0 ? function (row, index, isHead) { return ({}); } : _a, _b = props.getCellProps, getCellProps = _b === void 0 ? function (row, index, isHead) { return ({}); } : _b, rest = tslib_1.__rest(props, ["data", "headers", "getRowProps", "getCellProps"]);
    return (react_1.default.createElement(exports.DataTable, tslib_1.__assign({}, rest),
        react_1.default.createElement(exports.DataTableContent, null,
            !!headers && (react_1.default.createElement(exports.DataTableHead, null, headers.map(function (row, i) { return (react_1.default.createElement(exports.DataTableRow, tslib_1.__assign({ key: i }, getRowProps(row, i, true)), row.map(function (cell, j) { return (react_1.default.createElement(exports.DataTableHeadCell, tslib_1.__assign({ key: j }, getCellProps(cell, j, true)), cell)); }))); }))),
            react_1.default.createElement(exports.DataTableBody, null, data.map(function (row, i) { return (react_1.default.createElement(exports.DataTableRow, tslib_1.__assign({ key: i }, getRowProps(row, i, false)), row.map(function (cell, j) { return (react_1.default.createElement(exports.DataTableCell, tslib_1.__assign({ key: j }, getCellProps(cell, j, false)), cell)); }))); })))));
}
exports.SimpleDataTable = SimpleDataTable;
