// @rmwc/theme/dist/readme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.galleryExample = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var _doc_utils_1 = require("@doc-utils");
var generated_props_json_1 = tslib_1.__importDefault(require("./generated-props.json"));
var generated_examples_json_1 = tslib_1.__importDefault(require("./generated-examples.json"));
var _1 = require(".");
var button_1 = require("../button");
var radio_1 = require("../radio");
var checkbox_1 = require("../checkbox");
function default_1() {
    return (react_1.default.createElement(_doc_utils_1.Docs, { title: "Theming", lead: "MDC Theme is a foundational module that themes MDC Web components.", module: "@rmwc/theme", styles: ['@material/theme/dist/mdc.theme.css', '@rmwc/theme/theme.css'], docsLink: "https://material.io/develop/web/components/theme/", examples: generated_examples_json_1.default },
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Theme Options"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "The Theme module fully embraces using CSS variables for runtime theming. This allows for some really powerful usecases like a built in dark mode, custom palettes for your clients, or dynamic configuration for accessibility."),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Support for theming inside of `material-components-web` is not without issue, so RMWC maintains a theme fixes file to correct any anomalies for you. Please make sure you include both!"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "**Important** You should include the theme style sheets BEFORE any of your other styles."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement("div", { style: { backgroundColor: '#ddd' } }, [
                    'primary',
                    'secondary',
                    'error',
                    'background',
                    'surface',
                    'primaryBg',
                    'secondaryBg',
                    'textPrimaryOnBackground',
                    'textSecondaryOnBackground',
                    'textHintOnBackground',
                    'textDisabledOnBackground',
                    'textIconOnBackground',
                    'textPrimaryOnLight',
                    'textSecondaryOnLight',
                    'textHintOnLight',
                    'textDisabledOnLight',
                    'textIconOnLight'
                ].map(function (theme, i) { return (
                // @ts-ignore
                react_1.default.createElement(_1.Theme, { use: theme, key: i }, theme)); })),
                react_1.default.createElement("div", { style: { backgroundColor: '#333' } }, [
                    'onPrimary',
                    'onSecondary',
                    'onError',
                    'textPrimaryOnDark',
                    'textSecondaryOnDark',
                    'textHintOnDark',
                    'textDisabledOnDark',
                    'textIconOnDark'
                ].map(function (theme, i) { return (
                // @ts-ignore
                react_1.default.createElement(_1.Theme, { use: theme, key: i }, theme)); })))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "ThemeProvider"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "The `ThemeProvider` is an optional component that allows you to specify theme colors and settings for all of its subtree. This is useful to use once at the top of your app, or in parts of your app where the styles or color scheme differ."),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "You don't have to pass in all options. The `ThemeProvider` will automatically adjust some of the values like `onSurface` white or black text depending on colors contrast ratio."),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Theming in `material-components-web` isn't perfect, but a few basic options will get you most of the way. Try using the ThemePicker at the top and selecting \"Shrine\". You'll see that most things are colored appropriately, but the defaults provided for things like Buttons and tabs still have to have their colors overridden."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Defaults" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(button_1.Button, { raised: true }, "Cookies"),
                react_1.default.createElement(checkbox_1.Checkbox, { label: "Pizza", defaultChecked: true }),
                react_1.default.createElement(radio_1.Radio, { label: "Icecream", defaultChecked: true }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "With Provider" },
            react_1.default.createElement(_1.ThemeProvider, { options: {
                    primary: 'red',
                    secondary: 'blue'
                } },
                react_1.default.createElement(button_1.Button, { raised: true }, "Cookies"),
                react_1.default.createElement(checkbox_1.Checkbox, { label: "Pizza", defaultChecked: true }),
                react_1.default.createElement(radio_1.Radio, { label: "Icecream", defaultChecked: true }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "More Options" },
            react_1.default.createElement(_1.ThemeProvider, { options: {
                    primary: 'lightpink',
                    secondary: 'black',
                    onPrimary: '#000',
                    textPrimaryOnBackground: 'black'
                } },
                react_1.default.createElement(button_1.Button, { raised: true }, "Cookies"),
                react_1.default.createElement(checkbox_1.Checkbox, { label: "Pizza", defaultChecked: true }),
                react_1.default.createElement(radio_1.Radio, { label: "Icecream", defaultChecked: true }))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Theme Component"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "The Theme component allows you to apply theme colors to RMWC components, or components of your own. Almost every component in RMWC has a `theme` prop that you can use that takes the same options as the `Theme` component's `use` prop."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Custom" },
            react_1.default.createElement(_1.Theme, { use: ['primaryBg', 'onPrimary'], wrap: true },
                react_1.default.createElement("div", { style: { width: '4rem', height: '4rem', padding: '1rem' } }, "Cookies"))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Theme Prop" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Theme, { use: ['secondaryBg', 'onSecondary'], wrap: true },
                    react_1.default.createElement(button_1.Button, null, "Pizza")),
                react_1.default.createElement(button_1.Button, { theme: ['secondaryBg', 'onSecondary'] }, "Pizza"))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Typography" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement("h3", null,
                    "I ",
                    react_1.default.createElement(_1.Theme, { use: "primary" }, "Want"),
                    ' ',
                    react_1.default.createElement(_1.Theme, { use: "secondary" }, "Icecream")))),
        react_1.default.createElement(_doc_utils_1.DocProps, { src: generated_props_json_1.default, components: [
                { displayName: 'ThemeProvider', component: _1.ThemeProvider },
                { displayName: 'Theme', component: _1.Theme }
            ] })));
}
exports.default = default_1;
exports.galleryExample = (react_1.default.createElement(react_1.default.Fragment, null,
    react_1.default.createElement(_1.Theme, { style: {
            margin: '0.5rem',
            borderRadius: '6px',
            height: '3rem',
            width: '3rem',
            display: 'inline-block'
        }, use: "primaryBg" }),
    react_1.default.createElement(_1.Theme, { style: {
            margin: '0.5rem',
            borderRadius: '6px',
            height: '3rem',
            width: '3rem',
            display: 'inline-block'
        }, use: "secondaryBg" })));
