// @rmwc/tooltip/dist/readme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.galleryExample = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var _doc_utils_1 = require("@doc-utils");
var generated_props_json_1 = tslib_1.__importDefault(require("./generated-props.json"));
var generated_examples_json_1 = tslib_1.__importDefault(require("./generated-examples.json"));
var _1 = require(".");
var icon_button_1 = require("../icon-button");
var button_1 = require("../button");
var avatar_1 = require("../avatar");
var provider_1 = require("../provider");
function default_1() {
    return (react_1.default.createElement(_doc_utils_1.Docs, { title: "Tooltips", lead: "Tooltips display informative text when users hover over, focus on, or tap an element.", module: "@rmwc/tooltip", styles: ['@rmwc/tooltip/tooltip.css'], examples: generated_examples_json_1.default, addon: true },
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Basic Usage"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Wrap any component in a `Tooltip` and provide the content attribute. The only requirement is that is has a single React child, and that the child can accept `onMouseEnter`, `onMouseLeave`, `onFocus`, and `onClick` props."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Default" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Tooltip, { content: "Cookies" },
                    react_1.default.createElement(icon_button_1.IconButton, { icon: "star_border" })),
                react_1.default.createElement(_1.Tooltip, { content: "Pizza" },
                    react_1.default.createElement(icon_button_1.IconButton, { icon: "favorite_border" })),
                react_1.default.createElement(_1.Tooltip, { content: "Icecream" },
                    react_1.default.createElement(icon_button_1.IconButton, { icon: "mood" })))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Variants"),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "With Arrow" },
            react_1.default.createElement(_1.Tooltip, { content: "Cake", showArrow: true },
                react_1.default.createElement(icon_button_1.IconButton, { icon: "cake" }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Controlled / Always open" },
            react_1.default.createElement(_1.Tooltip, { content: "Hello", align: "right", open: true },
                react_1.default.createElement(icon_button_1.IconButton, { icon: "mood" }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Rich Content" },
            react_1.default.createElement(_1.Tooltip, { content: react_1.default.createElement("div", { style: { display: 'flex' } },
                    react_1.default.createElement(avatar_1.Avatar, { src: "images/avatars/captainamerica.png", size: "large", name: "Steve Rogers" }),
                    react_1.default.createElement("div", { style: { marginLeft: '0.5rem' } },
                        react_1.default.createElement("b", null, "Captain America"),
                        react_1.default.createElement("div", null, "Steve Rogers"))) },
                react_1.default.createElement("span", { role: "button" }, "S. Rogers"))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Styled content" },
            react_1.default.createElement(_1.Tooltip
            /** You make something like a popover window by just styling your inner content. */
            , { 
                /** You make something like a popover window by just styling your inner content. */
                content: react_1.default.createElement("div", { style: {
                        display: 'flex',
                        alignItems: 'center',
                        justifyContent: 'center',
                        background: 'white',
                        width: '20rem',
                        height: '8rem',
                        color: 'black',
                        borderRadius: '3px',
                        margin: '0 -3px'
                    } }, "Hello World!") },
                react_1.default.createElement("span", { role: "button" }, "Popover Window"))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Delay" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Tooltip, { content: "Cookies", enterDelay: 1000 },
                    react_1.default.createElement(button_1.Button, { label: "Enter Delay" })),
                react_1.default.createElement(_1.Tooltip, { content: "Pizza", leaveDelay: 1000 },
                    react_1.default.createElement(button_1.Button, { label: "Leave Delay" })),
                react_1.default.createElement(_1.Tooltip, { content: "Icecream", enterDelay: 1000, leaveDelay: 1000 },
                    react_1.default.createElement(button_1.Button, { label: "Both" })))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Alignment" }, function AlignmentExample() {
            return [
                'left',
                'right',
                'top',
                'bottom',
                'topLeft',
                'topRight',
                'bottomLeft',
                'bottomRight'
            ].map(function (align) { return (
            //@ts-ignore
            react_1.default.createElement(_1.Tooltip, { key: align, content: "Align: " + align, align: align },
                react_1.default.createElement(icon_button_1.IconButton, { icon: "trip_origin" }))); });
        }),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Activation"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "By default, tooltips will activate on hover and focus. You can change this behavior by passing one or more options to the `activateOn` prop."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Default" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Tooltip, { content: "Cookies", activateOn: "hover" },
                    react_1.default.createElement(button_1.Button, { label: "Hover" })),
                "''",
                react_1.default.createElement(_1.Tooltip, { content: "Pizza", activateOn: "click" },
                    react_1.default.createElement(button_1.Button, { label: "Click" })),
                react_1.default.createElement(_1.Tooltip, { content: "Icecream", activateOn: "focus" },
                    react_1.default.createElement(button_1.Button, { label: "Focus" })),
                react_1.default.createElement(_1.Tooltip, { content: "Cake", activateOn: ['hover', 'focus'] },
                    react_1.default.createElement(button_1.Button, { label: "Multiple" })))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Usage with RMWCProvider"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "The RMWCProvider allows you to specify global defaults for your tooltips."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Using Provider" },
            react_1.default.createElement(provider_1.RMWCProvider, { tooltip: {
                    align: 'right',
                    activateOn: 'hover',
                    showArrow: true,
                    leaveDelay: 500,
                    enterDelay: 0
                } },
                react_1.default.createElement(_1.Tooltip, { content: "Hello World!" },
                    react_1.default.createElement(button_1.Button, { label: "With Provider" })))),
        react_1.default.createElement(_doc_utils_1.DocProps, { src: generated_props_json_1.default, components: [{ displayName: 'Tooltip', component: _1.Tooltip }] })));
}
exports.default = default_1;
exports.galleryExample = (react_1.default.createElement(react_1.default.Fragment, null,
    react_1.default.createElement(_1.Tooltip, { content: "Favorite RMWC!", open: true },
        react_1.default.createElement(icon_button_1.IconButton, { icon: "favorite_outline" }))));
