// @rmwc/top-app-bar/dist/foundation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useTopAppBarFoundation = void 0;
var tslib_1 = require("tslib");
var react_1 = require("react");
var base_1 = require("@rmwc/base");
var top_app_bar_1 = require("@material/top-app-bar");
exports.useTopAppBarFoundation = function (props) {
    var scrollTargetRef = react_1.useRef(null);
    var navIconRef = react_1.useRef(null);
    var _a = base_1.useFoundation({
        props: props,
        elements: {
            rootEl: true
        },
        foundation: function (_a) {
            var rootEl = _a.rootEl, emit = _a.emit;
            var adapter = {
                hasClass: function (className) { return rootEl.hasClass(className); },
                addClass: function (className) { return rootEl.addClass(className); },
                removeClass: function (className) { return rootEl.removeClass(className); },
                setStyle: function (property, value) {
                    return rootEl.setStyle(property, value);
                },
                getTopAppBarHeight: function () { var _a; return ((_a = rootEl.ref) === null || _a === void 0 ? void 0 : _a.clientHeight) || 0; },
                notifyNavigationIconClicked: function () { return emit('onNav', {}); },
                getViewportScrollY: function () {
                    var target = scrollTargetRef.current;
                    return target
                        ? target['pageYOffset' in target ? 'pageYOffset' : 'scrollTop']
                        : 0;
                },
                getTotalActionItems: function () {
                    return rootEl.ref
                        ? rootEl.ref.querySelectorAll(top_app_bar_1.MDCTopAppBarFoundation.strings.ACTION_ITEM_SELECTOR).length
                        : 0;
                }
            };
            var foundation;
            if (props.short) {
                foundation = new top_app_bar_1.MDCShortTopAppBarFoundation(adapter);
            }
            else if (props.fixed) {
                foundation = new top_app_bar_1.MDCFixedTopAppBarFoundation(adapter);
            }
            else {
                foundation = new top_app_bar_1.MDCTopAppBarFoundation(adapter);
            }
            return foundation;
        }
    }), foundation = _a.foundation, elements = tslib_1.__rest(_a, ["foundation"]);
    var rootEl = elements.rootEl;
    react_1.useEffect(function () {
        var _a, _b;
        var target = props.scrollTarget || ((_b = (_a = rootEl.ref) === null || _a === void 0 ? void 0 : _a.ownerDocument) === null || _b === void 0 ? void 0 : _b.defaultView) || window;
        var handleTargetScroll = foundation.handleTargetScroll.bind(foundation);
        target.addEventListener('scroll', handleTargetScroll);
        scrollTargetRef.current = target;
        return function () {
            target.removeEventListener('scroll', handleTargetScroll);
        };
    }, [props.scrollTarget, scrollTargetRef, foundation, rootEl.ref]);
    react_1.useEffect(function () {
        var _a, _b;
        navIconRef.current =
            ((_a = rootEl.ref) === null || _a === void 0 ? void 0 : _a.querySelector(top_app_bar_1.MDCTopAppBarFoundation.strings.NAVIGATION_ICON_SELECTOR)) || null;
        var handler = foundation.handleNavigationClick.bind(foundation);
        (_b = navIconRef.current) === null || _b === void 0 ? void 0 : _b.addEventListener('click', handler);
        return function () {
            var _a;
            (_a = navIconRef.current) === null || _a === void 0 ? void 0 : _a.removeEventListener('click', handler);
        };
    }, [rootEl.ref, foundation]);
    // The Top App Bar sets these values in its constructor...
    // Reinit them after mount
    react_1.useEffect(function () {
        // @ts-ignore
        foundation.lastScrollPosition_ = foundation.adapter_.getViewportScrollY();
        // @ts-ignore
        foundation.topAppBarHeight_ = foundation.adapter_.getTopAppBarHeight();
    }, [foundation]);
    return tslib_1.__assign({ foundation: foundation }, elements);
};
