// @rmwc/menu/dist/readme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.galleryExample = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var _doc_utils_1 = require("@doc-utils");
var generated_props_json_1 = tslib_1.__importDefault(require("./generated-props.json"));
var generated_examples_json_1 = tslib_1.__importDefault(require("./generated-examples.json"));
var _1 = require(".");
var list_1 = require("../list");
var button_1 = require("../button");
var select_1 = require("../select");
var icon_button_1 = require("../icon-button");
var checkbox_1 = require("../checkbox");
function default_1() {
    return (react_1.default.createElement(_doc_utils_1.Docs, { title: "Menus", lead: "Menus display a list of choices on a transient sheet of material.", module: "@rmwc/menu", styles: [
            '@material/menu/dist/mdc.menu.css',
            '@material/menu-surface/dist/mdc.menu-surface.css',
            '@material/ripple/dist/mdc.ripple.css',
            '@material/list/dist/mdc.list.css',
            '@rmwc/icon/icon.css'
        ], docsLink: "https://material.io/develop/web/components/menus/", examples: generated_examples_json_1.default },
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Basic Usage"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "You can compose a menu with the given components, and manually manage the open state. `Menu` expects MenuItems as children while `MenuSurface` is a generic container which can have anything as a child."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null, function Example() {
            var _a = tslib_1.__read(react_1.default.useState(false), 2), open = _a[0], setOpen = _a[1];
            return (react_1.default.createElement(_1.MenuSurfaceAnchor, null,
                react_1.default.createElement(_1.Menu, { open: open, onSelect: function (evt) { return console.log(evt.detail.index); }, onClose: function (evt) { return setOpen(false); } },
                    react_1.default.createElement(_1.MenuItem, null, "Cookies"),
                    react_1.default.createElement(_1.MenuItem, null, "Pizza"),
                    react_1.default.createElement(list_1.ListDivider, null),
                    react_1.default.createElement(_1.MenuItem, null, "Icecream")),
                react_1.default.createElement(button_1.Button, { raised: true, onClick: function (evt) { return setOpen(!open); } }, "Menu")));
        }),
        react_1.default.createElement(_doc_utils_1.DocsExample, null, function Example() {
            var _a = tslib_1.__read(react_1.default.useState(false), 2), open = _a[0], setOpen = _a[1];
            return (react_1.default.createElement(_1.MenuSurfaceAnchor, null,
                react_1.default.createElement(_1.MenuSurface, { open: open, onClose: function (evt) { return setOpen(false); } },
                    react_1.default.createElement("div", { style: { padding: '1rem', width: '8rem' } }, "Make the content whatever you want.")),
                react_1.default.createElement(button_1.Button, { raised: true, onClick: function (evt) { return setOpen(!open); } }, "Menu Surface")));
        }),
        react_1.default.createElement(_doc_utils_1.DocsExample, null, function Example() {
            var _a = tslib_1.__read(react_1.default.useState(false), 2), open = _a[0], setOpen = _a[1];
            return (react_1.default.createElement(_1.MenuSurfaceAnchor, null,
                react_1.default.createElement(_1.MenuSurface, { open: open, onClose: function (evt) { return setOpen(false); } },
                    react_1.default.createElement("div", { style: { padding: '1rem', width: '8rem' } }, "Menu")),
                react_1.default.createElement(icon_button_1.IconButton, { icon: "menu", onClick: function (evt) { return setOpen(!open); } })));
        }),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Simplified usage"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "RMWC provides a convenience `SimpleMenu` component that takes a handle as a prop, and manages the open state for you."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(_1.SimpleMenu, { handle: react_1.default.createElement(button_1.Button, null, "Simple Menu") },
                react_1.default.createElement(_1.MenuItem, null, "Cookies"),
                react_1.default.createElement(_1.MenuItem, null, "Pizza"),
                react_1.default.createElement(_1.MenuItem, null, "Icecream"))),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(_1.SimpleMenuSurface, { handle: react_1.default.createElement(button_1.Button, null, "Simple Menu Surface") },
                react_1.default.createElement("div", { style: { padding: '1rem', width: '8rem' } }, "Make the content whatever you want."))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Anchoring"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "By default, Menus will attempt to automatically position themselves, but this behavior can be overridden by setting the `anchorCorner` prop."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null, function Example() {
            var _a = tslib_1.__read(react_1.default.useState('topLeft'), 2), anchorCorner = _a[0], setAnchorCorner = _a[1];
            return (react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.MenuSurfaceAnchor, null,
                    react_1.default.createElement(_1.MenuSurface, { anchorCorner: anchorCorner, open: true },
                        react_1.default.createElement("div", { style: { padding: '1rem', width: '8rem' } },
                            "anchorCorner: ",
                            anchorCorner)),
                    react_1.default.createElement(button_1.Button, { raised: true, label: "Anchored Menu" })),
                react_1.default.createElement(select_1.Select, { value: anchorCorner, label: "anchorCorner", onChange: function (evt) { return setAnchorCorner(evt.currentTarget.value); }, options: [
                        'topLeft',
                        'topRight',
                        'bottomLeft',
                        'bottomRight',
                        'topStart',
                        'topEnd',
                        'bottomStart',
                        'bottomEnd'
                    ] })));
        }),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Rendering through Portals"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Occasionally, you may find your menu being cut off from being inside a container that is styled to be `overflow:hidden`. RMWC provides a `renderToPortal` prop that lets you use React's portal functionality to render the menu dropdown in a different container."),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "You can specify any element or selector you want, but the simplest method is to pass `true` and use RMWC's built in `Portal` component."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { codeOnly: true }, /* jsx */ "\n          // Somewhere at the top level of your app\n          // Render the RMWC Portal\n          // You only have to do this once\n          import React from 'react';\n          import { Portal } from '@rmwc/base';\n\n          export default function App() {\n            return (\n              <div>\n                ...\n                <Portal />\n              </div>\n            )\n          }\n        "),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Now you can use the `renderToPortal` prop. Below is a contrived example of a menu being cut off due to `overflow: hidden`."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null, function Example() {
            var _a = tslib_1.__read(react_1.default.useState(true), 2), renderToPortal = _a[0], setRenderToPortal = _a[1];
            var options = ['Cookies', 'Pizza', 'Icecream'];
            return (react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement("div", { style: {
                        marginBottom: '10rem',
                        height: '3.5rem',
                        overflow: 'hidden'
                    } },
                    react_1.default.createElement(_1.MenuSurfaceAnchor, null,
                        react_1.default.createElement(button_1.Button, { raised: true }, "Open Menu"),
                        react_1.default.createElement(_1.Menu, { open: true, renderToPortal: renderToPortal }, options.map(function (o) { return (react_1.default.createElement(_1.MenuItem, { key: o }, o)); })))),
                react_1.default.createElement(checkbox_1.Checkbox, { checked: renderToPortal, onChange: function (evt) { return setRenderToPortal(evt.currentTarget.checked); }, label: "renderToPortal" })));
        }),
        react_1.default.createElement(_doc_utils_1.DocProps, { src: generated_props_json_1.default, components: [
                { displayName: 'Menu', component: _1.Menu },
                { displayName: 'MenuItem', component: _1.MenuItem },
                { displayName: 'MenuSurface', component: _1.MenuSurface },
                { displayName: 'MenuSurfaceAnchor', component: _1.MenuSurfaceAnchor },
                { displayName: 'SimpleMenu', component: _1.SimpleMenu },
                { displayName: 'SimpleMenuSurface', component: _1.SimpleMenuSurface }
            ] })));
}
exports.default = default_1;
exports.galleryExample = (react_1.default.createElement("div", { "aria-hidden": "false", className: "mdc-menu  mdc-menu-surface--open mdc-menu-surface", style: { position: 'static' } },
    react_1.default.createElement("div", { role: "menu", className: "mdc-list mdc-menu__items mdc-list" },
        react_1.default.createElement("div", { role: "menuitem", className: "mdc-ripple-upgraded mdc-list-item" }, "Cookies"),
        react_1.default.createElement("div", { role: "menuitem", className: "mdc-ripple-upgraded mdc-list-item" }, "Pizza"),
        react_1.default.createElement("div", { role: "menuitem", className: "mdc-ripple-upgraded mdc-list-item" }, "Icecream"))));
