// rmwc/dist/docs/views/home/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Home = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var react_router_dom_1 = require("react-router-dom");
var package_json_1 = require("../../../package.json");
var typography_1 = require("@rmwc/typography");
var card_1 = require("@rmwc/card");
var grid_1 = require("@rmwc/grid");
var theme_1 = require("@rmwc/theme");
var icon_1 = require("@rmwc/icon");
var home_module_css_1 = tslib_1.__importDefault(require("./home.module.css"));
var button_1 = require("@rmwc/button");
var menu_content_1 = require("../../common/menu-content");
var checklist = [
    "Uses Google's official material-components-web library",
    'Includes additional addon components not offered by Google.',
    'Works in React from 16.8.x and up',
    'First class Typescript Support',
    'Server side rendering support',
    'Individually packaged and released components'
];
function Example(_a) {
    var ex = _a.ex, name = _a.name, url = _a.url;
    return (react_1.default.createElement(react_router_dom_1.Link, { to: url, className: home_module_css_1.default.example, onClick: function () { return window.scrollTo(0, 0); } },
        react_1.default.createElement("div", { className: home_module_css_1.default.exampleInner },
            react_1.default.createElement("div", null, ex)),
        react_1.default.createElement("div", { className: home_module_css_1.default.exampleName },
            react_1.default.createElement(typography_1.Typography, { use: "headline6" }, name))));
}
exports.Home = function () {
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("header", { className: home_module_css_1.default.header },
            react_1.default.createElement("div", { className: "container" },
                react_1.default.createElement(theme_1.Theme, { use: ['onPrimary'] },
                    react_1.default.createElement("div", { className: home_module_css_1.default.intro },
                        react_1.default.createElement(typography_1.Typography, { use: "headline3", tag: "h1", className: home_module_css_1.default.headline1 }, "React Material Web Components"),
                        react_1.default.createElement(typography_1.Typography, { use: "headline6", tag: "h2", className: home_module_css_1.default.headline2 }, "A React UI Kit built on Google's official Material Components Web library"),
                        react_1.default.createElement(react_router_dom_1.Link, { to: "/installation" },
                            react_1.default.createElement(button_1.Button, { theme: ['secondaryBg', 'onSecondary'], raised: true }, "Get Started")),
                        react_1.default.createElement(typography_1.Typography, { tag: "h3", use: "caption" },
                            "v",
                            package_json_1.version),
                        react_1.default.createElement("div", { className: home_module_css_1.default.checklist }, checklist.map(function (c) { return (react_1.default.createElement("div", { key: c },
                            react_1.default.createElement(icon_1.Icon, { theme: "secondary", icon: "check" }),
                            " ",
                            react_1.default.createElement("div", null, c))); })))),
                react_1.default.createElement(grid_1.Grid, { className: home_module_css_1.default.featureGrid },
                    react_1.default.createElement(grid_1.GridCell, { span: 4 },
                        react_1.default.createElement(react_router_dom_1.Link, { to: "/installation" },
                            react_1.default.createElement(card_1.Card, { theme: ['primaryBg', 'onPrimary'] },
                                react_1.default.createElement("div", { style: { padding: '1rem' } },
                                    react_1.default.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Simple to Use"),
                                    react_1.default.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Get started in 5 minutes.")),
                                react_1.default.createElement(card_1.CardMedia, { style: {
                                        backgroundSize: 'fill',
                                        backgroundImage: 'url(images/backgrounds/home-1.png)',
                                        minHeight: '5rem',
                                        flex: 1
                                    } })))),
                    react_1.default.createElement(grid_1.GridCell, { span: 4 },
                        react_1.default.createElement(card_1.Card, { theme: "background", tag: "a", href: "https://material.io/components/web/" },
                            react_1.default.createElement("div", { style: { padding: '1rem' } },
                                react_1.default.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Material Accuracy"),
                                react_1.default.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Uses Google's official Material Components Web library.")),
                            react_1.default.createElement(card_1.CardMedia, { style: {
                                    backgroundSize: 'cover',
                                    backgroundImage: 'url(images/backgrounds/home-2.png)',
                                    minHeight: '5rem',
                                    flex: 1
                                } }))),
                    react_1.default.createElement(grid_1.GridCell, { span: 4 },
                        react_1.default.createElement(react_router_dom_1.Link, { to: "/styling-theming" },
                            react_1.default.createElement(card_1.Card, { style: { backgroundColor: '#212121' }, theme: "textPrimaryOnDark" },
                                react_1.default.createElement("div", { style: { padding: '1rem' } },
                                    react_1.default.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Easy to Customize"),
                                    react_1.default.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Integrate with a variety of CSS workflows.")),
                                react_1.default.createElement(card_1.CardMedia, { style: {
                                        backgroundSize: 'cover',
                                        backgroundImage: 'url(images/backgrounds/home-3.png)',
                                        minHeight: '5rem',
                                        flex: 1
                                    } }))))))),
        react_1.default.createElement("div", { className: home_module_css_1.default.examples }, menu_content_1.galleryContent.map(function (ex) {
            return (react_1.default.createElement(Example, { key: ex.label, name: ex.label, ex: ex.gallery, url: ex.url }));
        }))));
};
exports.default = exports.Home;
