// rmwc/dist/docs/views/site-search/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteSearch = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var menu_content_1 = require("../../common/menu-content");
var menu_1 = require("@rmwc/menu");
var list_1 = require("@rmwc/list");
var textfield_1 = require("@rmwc/textfield");
var react_router_dom_1 = require("react-router-dom");
var circular_progress_1 = require("@rmwc/circular-progress");
var history_1 = require("../../common/history");
var site_search_module_css_1 = tslib_1.__importDefault(require("./site-search.module.css"));
var componentsList = (function () {
    var walkOptions = function (options) {
        return options.reduce(function (acc, val) {
            if (val.options) {
                acc.push.apply(acc, tslib_1.__spread(walkOptions(val.options)));
            }
            else {
                acc.push(val);
            }
            return acc;
        }, []);
    };
    return walkOptions(menu_content_1.menuContent);
})();
var searchComponents = function (val) {
    return componentsList
        .filter(function (c) {
        return c.label.toLowerCase().includes(val.toLowerCase());
    })
        .map(function (c) { return ({
        id: c.label,
        icon: {
            icon: 'code',
            theme: 'primary'
        },
        sectionName: c.label,
        snippet: "View docs for " + c.label,
        url: c.url
    }); });
};
var searchGoogle = function (val, abortController) { return tslib_1.__awaiter(void 0, void 0, void 0, function () {
    var _a, items;
    return tslib_1.__generator(this, function (_b) {
        switch (_b.label) {
            case 0: return [4 /*yield*/, fetch("https://www.googleapis.com/customsearch/v1/siterestrict?key=" + process.env.REACT_APP_CUSTOM_SEARCH_KEY + "&cx=" + process.env.REACT_APP_CUSTOM_SEARCH_ID + "&q=" + val, { signal: abortController.signal }).then(function (res) { return res.json(); })];
            case 1:
                _a = (_b.sent()).items, items = _a === void 0 ? [] : _a;
                return [2 /*return*/, (items
                        // shitty hack to ignore things in the index that just reference them homepage
                        .filter(function (r) { return !r.snippet.startsWith('RMWC is a React wrapper'); })
                        .map(function (r) { return ({
                        id: r.cacheId,
                        icon: {
                            icon: 'notes',
                            theme: 'primary'
                        },
                        sectionName: (r.title.split('|').pop() || '').trim(),
                        snippet: r.snippet,
                        // resolve from the full url into the page
                        url: ((r.formattedUrl.split('?').pop() || '')
                            .split('&')
                            .find(function (p) { return p.startsWith('p='); }) || '').slice(2)
                    }); }))];
        }
    });
}); };
function SiteSearch() {
    var _this = this;
    var _a = tslib_1.__read(react_1.useState(''), 2), searchVal = _a[0], _setSearchVal = _a[1];
    var _b = tslib_1.__read(react_1.useState(false), 2), isSearching = _b[0], setIsSearching = _b[1];
    var timerIdRef = react_1.useRef();
    var abortControllerRef = react_1.useRef();
    var _c = tslib_1.__read(react_1.useState([]), 2), results = _c[0], setResults = _c[1];
    var doSearch = function (val) {
        if (abortControllerRef.current) {
            abortControllerRef.current.abort();
            abortControllerRef.current = undefined;
        }
        if (timerIdRef.current) {
            clearTimeout(timerIdRef.current);
        }
        var abortController = new AbortController();
        abortControllerRef.current = abortController;
        timerIdRef.current = window.setTimeout(function () { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            var components, searchResults, err_1;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        components = searchComponents(val);
                        setResults(components);
                        _a.label = 1;
                    case 1:
                        _a.trys.push([1, 3, , 4]);
                        return [4 /*yield*/, searchGoogle(val, abortController)];
                    case 2:
                        searchResults = _a.sent();
                        abortControllerRef.current = undefined;
                        setResults(tslib_1.__spread(components, searchResults));
                        return [3 /*break*/, 4];
                    case 3:
                        err_1 = _a.sent();
                        console.warn(err_1);
                        abortControllerRef.current = undefined;
                        return [3 /*break*/, 4];
                    case 4: return [2 /*return*/];
                }
            });
        }); }, 200);
    };
    var setSearchVal = function (val) {
        if (val.length > 2) {
            !isSearching && setIsSearching(true);
            doSearch(val);
        }
        else {
            setResults([]);
        }
        _setSearchVal(val);
    };
    return (react_1.default.createElement(menu_1.MenuSurfaceAnchor, { className: site_search_module_css_1.default.siteSearchWrapper },
        react_1.default.createElement(textfield_1.TextField, { placeholder: "Search", icon: "search", trailingIcon: {
                icon: searchVal ? 'close' : '',
                onClick: function () { return setSearchVal(''); }
            }, outlined: true, className: site_search_module_css_1.default.siteSearch, value: searchVal, onChange: function (evt) { return setSearchVal(evt.currentTarget.value); }, onFocus: function () { return setIsSearching(true); }, onKeyDown: function (evt) {
                if (evt.which === 40) {
                    var listItem = document.querySelector("." + site_search_module_css_1.default.siteSearchMenu + " a");
                    if (isSearching && listItem) {
                        listItem.focus();
                    }
                }
            } }),
        react_1.default.createElement(menu_1.Menu, { className: site_search_module_css_1.default.siteSearchMenu, open: isSearching && searchVal.length > 2, anchorCorner: "bottomStart", focusOnOpen: false, onFocus: function () { return setIsSearching(true); }, onClose: function () {
                setIsSearching(false);
            }, onSelect: function (evt) {
                window.scrollTo(0, 0);
                history_1.history.replace(evt.detail.item.href.split('/').pop() || '/');
                document.activeElement &&
                    document.activeElement.blur();
            } },
            react_1.default.createElement(menu_1.MenuItems, { twoLine: true },
                results.map(function (r) { return (react_1.default.createElement(list_1.SimpleListItem, { tag: react_router_dom_1.Link, to: r.url, graphic: r.icon, key: r.id, text: r.sectionName, secondaryText: r.snippet })); }),
                !!abortControllerRef.current && (react_1.default.createElement("div", { className: site_search_module_css_1.default.loading },
                    react_1.default.createElement(circular_progress_1.CircularProgress, { size: "large" }))),
                !abortControllerRef.current && !results.length && (react_1.default.createElement(list_1.SimpleListItem, { text: "No results found" }))))));
}
exports.SiteSearch = SiteSearch;
