// @rmwc/image-list/dist/readme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.galleryExample = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var _doc_utils_1 = require("@doc-utils");
var generated_props_json_1 = tslib_1.__importDefault(require("./generated-props.json"));
var generated_examples_json_1 = tslib_1.__importDefault(require("./generated-examples.json"));
var _1 = require(".");
function default_1() {
    return (react_1.default.createElement(_doc_utils_1.Docs, { title: "Image Lists", lead: "MDC Image List provides a RTL-aware Material Design image list component, representing an evolution of the Material Design Grid List spec. An Image List consists of several items, each containing an image and optionally supporting content (i.e. a text label).", module: "@rmwc/image-list", styles: ['@material/image-list/dist/mdc.image-list.css'], docsLink: "https://material.io/develop/web/components/image-lists/", examples: generated_examples_json_1.default },
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Standard Layout"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Image Lists will give you basic layout, but you will have to use inline styling or CSS to achieve things like spacing, aspect ratio, and column count. Check out the inline styles below as examples."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Default" },
            react_1.default.createElement(_1.ImageList, null, [
                'images/photos/3x2-1.jpg',
                'images/photos/2x3-1.jpg',
                'images/photos/3x2-4.jpg',
                'images/photos/3x2-5.jpg',
                'images/photos/3x2-6.jpg',
                'images/photos/2x3-2.jpg',
                'images/photos/3x2-8.jpg',
                'images/photos/3x2-11.jpg',
                'images/photos/2x3-5.jpg',
                'images/photos/3x2-13.jpg',
                'images/photos/3x2-14.jpg',
                'images/photos/2x3-6.jpg',
                'images/photos/3x2-15.jpg',
                'images/photos/3x2-16.jpg',
                'images/photos/2x3-7.jpg'
            ].map(function (src) { return (react_1.default.createElement(_1.ImageListItem, { key: src, style: { margin: '2px', width: 'calc(100% / 5 - 4.2px)' } },
                react_1.default.createElement(_1.ImageListImageAspectContainer, { style: { paddingBottom: 'calc(100% / 1.5)' } },
                    react_1.default.createElement(_1.ImageListImage, { src: src })),
                react_1.default.createElement(_1.ImageListSupporting, null,
                    react_1.default.createElement(_1.ImageListLabel, null, "Text label")))); }))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Masonry Layout"),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(_1.ImageList, { masonry: true, withTextProtection: true, style: {
                    columnCount: 5,
                    columnGap: '16px',
                    maxWidth: '1000px'
                } }, [
                'images/photos/3x2-1.jpg',
                'images/photos/2x3-1.jpg',
                'images/photos/3x2-4.jpg',
                'images/photos/3x2-5.jpg',
                'images/photos/2x3-5.jpg',
                'images/photos/3x2-6.jpg',
                'images/photos/2x3-2.jpg',
                'images/photos/3x2-8.jpg',
                'images/photos/3x2-11.jpg',
                'images/photos/3x2-13.jpg',
                'images/photos/3x2-14.jpg',
                'images/photos/2x3-6.jpg',
                'images/photos/3x2-15.jpg',
                'images/photos/2x3-7.jpg',
                'images/photos/3x2-16.jpg'
            ].map(function (src) { return (react_1.default.createElement(_1.ImageListItem, { key: src, style: { marginBottom: '16px' } },
                react_1.default.createElement(_1.ImageListImage, { src: src }),
                react_1.default.createElement(_1.ImageListSupporting, null,
                    react_1.default.createElement(_1.ImageListLabel, null, "Text label")))); }))),
        react_1.default.createElement(_doc_utils_1.DocProps, { src: generated_props_json_1.default, components: [
                { displayName: 'ImageList', component: _1.ImageList },
                { displayName: 'ImageListItem', component: _1.ImageListItem },
                {
                    displayName: 'ImageListImageAspectContainer',
                    component: _1.ImageListImageAspectContainer
                },
                { displayName: 'ImageListImage', component: _1.ImageListImage },
                {
                    displayName: 'ImageListSupporting',
                    component: _1.ImageListSupporting
                },
                { displayName: 'ImageListLabel', component: _1.ImageListLabel }
            ] })));
}
exports.default = default_1;
exports.galleryExample = (react_1.default.createElement(_1.ImageList, { masonry: true, withTextProtection: true, style: {
        transform: 'scale(0.5)',
        columnCount: 3,
        columnGap: '16px',
        maxWidth: '1000px'
    } }, [
    'images/photos/3x2-1.jpg',
    'images/photos/2x3-1.jpg',
    'images/photos/3x2-4.jpg',
    'images/photos/3x2-5.jpg',
    'images/photos/2x3-5.jpg',
    'images/photos/3x2-6.jpg',
    'images/photos/2x3-2.jpg',
    'images/photos/3x2-8.jpg',
    'images/photos/3x2-11.jpg',
    'images/photos/3x2-13.jpg',
    'images/photos/3x2-14.jpg'
].map(function (src) { return (react_1.default.createElement(_1.ImageListItem, { key: src, style: { marginBottom: '16px' } },
    react_1.default.createElement(_1.ImageListImage, { src: src }))); })));
