// @rmwc/radio/dist/foundation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useRadioFoundation = void 0;
var tslib_1 = require("tslib");
var toggleable_1 = require("@rmwc/toggleable");
var base_1 = require("@rmwc/base");
var radio_1 = require("@material/radio");
exports.useRadioFoundation = function (props) {
    var _a = toggleable_1.useToggleFoundation(props), renderToggle = _a.renderToggle, toggleRootProps = _a.toggleRootProps, id = _a.id;
    var _b = base_1.useFoundation({
        props: props,
        elements: {
            rootEl: true
        },
        foundation: function (_a) {
            var rootEl = _a.rootEl;
            return new radio_1.MDCRadioFoundation({
                addClass: function (className) { return rootEl.addClass(className); },
                removeClass: function (className) { return rootEl.removeClass(className); }
            });
        }
    }), foundation = _b.foundation, elements = tslib_1.__rest(_b, ["foundation"]);
    return tslib_1.__assign({ foundation: foundation, renderToggle: renderToggle, toggleRootProps: toggleRootProps, id: id }, elements);
};
