// @rmwc/radio/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Radio = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var base_1 = require("@rmwc/base");
var ripple_1 = require("@rmwc/ripple");
var foundation_1 = require("./foundation");
/** A Radio button component. */
exports.Radio = base_1.createComponent(function Radio(props, ref) {
    var _a = foundation_1.useRadioFoundation(props), renderToggle = _a.renderToggle, id = _a.id, toggleRootProps = _a.toggleRootProps, rootEl = _a.rootEl;
    var children = props.children, className = props.className, label = props.label, style = props.style, inputRef = props.inputRef, foundationRef = props.foundationRef, rest = tslib_1.__rest(props, ["children", "className", "label", "style", "inputRef", "foundationRef"]);
    var radio = (react_1.default.createElement(RadioRoot, tslib_1.__assign({}, rootEl.props(toggleRootProps), { ref: base_1.mergeRefs(rootEl.setRef, ref) }),
        react_1.default.createElement("input", tslib_1.__assign({}, rest, { className: "mdc-radio__native-control", type: "radio", id: id, ref: inputRef })),
        react_1.default.createElement(RadioBackground, null),
        react_1.default.createElement(RadioRipple, null)));
    return renderToggle(radio);
});
/*********************************************************************
 * Bits
 *********************************************************************/
var RadioRipple = react_1.default.memo(function RadioRipple() {
    return react_1.default.createElement("div", { className: "mdc-radio__ripple" });
});
var RadioRoot = ripple_1.withRipple({
    surface: false,
    unbounded: true
})(react_1.default.forwardRef(function RadioRoot(props, ref) {
    var disabled = props.disabled, rest = tslib_1.__rest(props, ["disabled"]);
    var className = base_1.useClassNames(props, [
        'mdc-radio',
        {
            'mdc-radio--disabled': disabled
        }
    ]);
    return react_1.default.createElement(base_1.Tag, tslib_1.__assign({}, rest, { className: className, ref: ref }));
}));
var RadioBackground = react_1.default.memo(function RadioBackground() {
    return (react_1.default.createElement("div", { className: "mdc-radio__background" },
        react_1.default.createElement("div", { className: "mdc-radio__outer-circle" }),
        react_1.default.createElement("div", { className: "mdc-radio__inner-circle" })));
});
