// @rmwc/badge/dist/readme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.galleryExample = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var _doc_utils_1 = require("@doc-utils");
var generated_props_json_1 = tslib_1.__importDefault(require("./generated-props.json"));
var generated_examples_json_1 = tslib_1.__importDefault(require("./generated-examples.json"));
var _1 = require(".");
var button_1 = require("../button");
var icon_button_1 = require("../icon-button");
var avatar_1 = require("../avatar");
function default_1() {
    return (react_1.default.createElement(_doc_utils_1.Docs, { title: "Badges", lead: "Badges are small status descriptors for UI elements. A badge consists of a small circle, typically containing a number or other short set of characters, that appears in proximity to another object.", module: "@rmwc/badge", styles: ['@rmwc/badge/badge.css'], examples: generated_examples_json_1.default, addon: true },
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Basic" },
            react_1.default.createElement(_1.Badge, { align: "inline" })),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Labels" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Badge, { align: "inline", label: 20 }),
                react_1.default.createElement(_1.Badge, { align: "inline", label: "99+" }),
                react_1.default.createElement(_1.Badge, { align: "inline", label: "New" }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, { label: "Theming" },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.Badge, { theme: ['primaryBg', 'onPrimary'], align: "inline" }),
                react_1.default.createElement(_1.Badge, { style: { background: 'hotpink' }, align: "inline" }),
                react_1.default.createElement(_1.Badge, { theme: ['secondaryBg', 'onSecondary'], align: "inline", label: "Theme" }))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Usage with other components"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "The badge component has been designed to play well with the majority of components in RMWC. You can place it inside of any component that accepts children and its default position will be absolute to the top end corner."),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Because passing a Badge as a child doesn't always work (for things like `overflow: hidden` elements), you can use the `BadgeAnchor` component. This is really just a div with `position: relative` and some other sensible layout properties set on it, so you can use this or your own CSS to achieve the same result. Additionally, exact positioning is highly dependent on your design and shape of your components. Badges provide an `inset` property that allows you to adjust the positioning of the Badge as necessary."),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(button_1.Button, { raised: true, label: "Button" }),
                    react_1.default.createElement(_1.Badge, null)),
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(button_1.Button, { raised: true, label: "Button", theme: ['secondaryBg', 'onSecondary'] }),
                    react_1.default.createElement(_1.Badge, { style: { background: 'hotpink' }, label: "Hello" })))),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(_1.BadgeAnchor, null,
                react_1.default.createElement(icon_button_1.IconButton, { icon: "notifications" }),
                react_1.default.createElement(_1.Badge, { inset: "0.75rem" }))),
        react_1.default.createElement(_doc_utils_1.DocsExample, null,
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(avatar_1.Avatar, { src: "images/avatars/ironman.png", size: "large", name: "Tony Stark" }),
                    react_1.default.createElement(_1.Badge, { inset: "5px" })),
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(avatar_1.Avatar, { src: "images/avatars/blackwidow.png", size: "large", name: "Natalia Alianovna Romanova", square: true }),
                    react_1.default.createElement(_1.Badge, null)))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Alignment"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "Badges can be aligned to the start, end, or use inline alignment. They are also RTL aware. They default to align end."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { center: true },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(button_1.Button, { raised: true, label: "Align Start" }),
                    react_1.default.createElement(_1.Badge, { align: "start" })),
                react_1.default.createElement(_1.BadgeAnchor, null,
                    react_1.default.createElement(button_1.Button, { raised: true, label: "Align End" }),
                    react_1.default.createElement(_1.Badge, { align: "end" })))),
        react_1.default.createElement(_doc_utils_1.DocsSubtitle, null, "Transitions"),
        react_1.default.createElement(_doc_utils_1.DocsP, null, "You can transition between the standalone indicator and a badge with content. The badge will consider any `label` other than null or undefined as valid content."),
        react_1.default.createElement(_doc_utils_1.DocsExample, { center: true }, function Example() {
            var _a = tslib_1.__read(react_1.default.useState(undefined), 2), label = _a[0], setLabel = _a[1];
            react_1.default.useEffect(function () {
                var timeout = setTimeout(function () {
                    switch (label) {
                        case '99+':
                            setLabel(undefined);
                            break;
                        case '':
                            setLabel('99+');
                            break;
                        case undefined:
                            setLabel('');
                            break;
                    }
                }, 1800);
                return function () { return clearTimeout(timeout); };
            }, [label]);
            return (react_1.default.createElement(_1.BadgeAnchor, null,
                react_1.default.createElement(button_1.Button, { raised: true, label: "Button" }),
                react_1.default.createElement(_1.Badge, { label: label, exited: label === undefined })));
        }),
        react_1.default.createElement(_doc_utils_1.DocProps, { src: generated_props_json_1.default, components: [
                { displayName: 'Badge', component: _1.Badge },
                { displayName: 'BadgeAnchor', component: _1.BadgeAnchor }
            ] })));
}
exports.default = default_1;
exports.galleryExample = (react_1.default.createElement(_1.BadgeAnchor, { style: {
        width: '3rem',
        height: '3rem',
        border: '1px solid rgba(0,0,0,.33)',
        borderRadius: '0.5rem'
    } },
    react_1.default.createElement(_1.Badge, { label: "+99" })));
