// @rmwc/circular-progress/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CircularProgress = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var base_1 = require("@rmwc/base");
var SIZE_MAP = {
    xsmall: 18,
    small: 20,
    medium: 24,
    large: 36,
    xlarge: 48
};
exports.CircularProgress = base_1.createComponent(function CircularProgress(props, ref) {
    var _a;
    var _b = props.size, size = _b === void 0 ? 'medium' : _b, _c = props.max, max = _c === void 0 ? 1 : _c, _d = props.min, min = _d === void 0 ? 0 : _d, progress = props.progress, rest = tslib_1.__rest(props, ["size", "max", "min", "progress"]);
    var className = base_1.useClassNames(props, [
        'rmwc-circular-progress',
        (_a = {},
            _a["rmwc-circular-progress--size-" + props.size] = typeof props.size === 'string',
            _a['rmwc-circular-progress--indeterminate'] = progress === undefined,
            _a['rmwc-circular-progress--thickerstroke'] = !!props.size && (SIZE_MAP[size] || Number(size)) > 36,
            _a)
    ]);
    var style = !SIZE_MAP[size]
        ? tslib_1.__assign(tslib_1.__assign({}, rest.style), { fontSize: Number(size) }) : rest.style;
    var _size = SIZE_MAP[size] || Number(size);
    var calculateRatio = function (value) {
        if (value < min)
            return 0;
        if (value > max)
            return 1;
        return (value - min) / (max - min);
    };
    var circularStyle = function (size) {
        return progress !== undefined
            ? {
                strokeDasharray: 2 * Math.PI * (size / 2.4) * calculateRatio(progress) + ", 666.66%"
            }
            : undefined;
    };
    return (react_1.default.createElement(base_1.Tag, tslib_1.__assign({ "aria-valuenow": progress, "aria-valuemin": min, "aria-valuemax": max }, rest, { style: style, className: className, ref: ref }),
        react_1.default.createElement("svg", { className: "rmwc-circular-progress__circle", viewBox: "0 0 " + _size + " " + _size },
            react_1.default.createElement("circle", { className: "rmwc-circular-progress__path", style: circularStyle(_size), cx: _size / 2, cy: _size / 2, r: _size / 2.4 }))));
});
