// rmwc/dist/docs/views/app/theme-picker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTheme = exports.ThemePicker = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var tabs_1 = require("@rmwc/tabs");
var list_1 = require("@rmwc/list");
var menu_1 = require("@rmwc/menu");
var button_1 = require("@rmwc/button");
var base_1 = require("@rmwc/base");
var utils_1 = require("@rmwc/theme/utils");
var top_app_bar_1 = require("@rmwc/top-app-bar");
var DEFAULT_THEME = {
    '--mdc-theme-primary': '#6200ee',
    '--mdc-theme-secondary': '#03dac4',
    '--mdc-theme-background': '#fff',
    '--mdc-theme-surface': '#fff',
    '--mdc-theme-error': '#b00020'
};
var TEXT_DEFAULTS = {
    '--mdc-theme-on-primary': '#fff',
    '--mdc-theme-on-secondary': '#fff',
    '--mdc-theme-on-surface': '#000',
    '--mdc-theme-on-error': '#fff',
    '--mdc-theme-text-primary-on-background': 'rgba(0, 0, 0, 0.87)',
    '--mdc-theme-text-secondary-on-background': 'rgba(0, 0, 0, 0.54)',
    '--mdc-theme-text-hint-on-background': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-disabled-on-background': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-icon-on-background': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-primary-on-light': 'rgba(0, 0, 0, 0.87)',
    '--mdc-theme-text-secondary-on-light': 'rgba(0, 0, 0, 0.54)',
    '--mdc-theme-text-hint-on-light': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-disabled-on-light': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-icon-on-light': 'rgba(0, 0, 0, 0.38)',
    '--mdc-theme-text-primary-on-dark': 'white',
    '--mdc-theme-text-secondary-on-dark': 'rgba(255, 255, 255, 0.7)',
    '--mdc-theme-text-hint-on-dark': 'rgba(255, 255, 255, 0.5)',
    '--mdc-theme-text-disabled-on-dark': 'rgba(255, 255, 255, 0.5)',
    '--mdc-theme-text-icon-on-dark': 'rgba(255, 255, 255, 0.5)'
};
var THEMES = {
    Baseline: {
        '--mdc-theme-primary': '#6200ee',
        '--mdc-theme-secondary': '#03dac4' /** Any theme option pointing to a valid CSS value. */
    },
    Crane: {
        '--mdc-theme-primary': '#5d1049',
        '--mdc-theme-secondary': '#fa3336'
    },
    Fortnightly: {
        '--mdc-theme-primary': '#303030',
        '--mdc-theme-secondary': '#661fff'
    },
    Shrine: {
        '--mdc-theme-primary': '#ffdbcf',
        '--mdc-theme-secondary': '#feeae6'
    },
    Dark: {
        '--mdc-theme-background': '#212121',
        '--mdc-theme-surface': '#37474F',
        '--mdc-theme-on-surface': 'rgba(255,255,255,.87)',
        '--mdc-theme-primary': '#24aee9',
        '--mdc-theme-on-primary': 'rgba(255,255,255,.87)',
        '--mdc-theme-secondary': '#e539ff',
        '--mdc-theme-on-secondary': 'rgba(0,0,0,0.87)'
    }
};
var ThemePicker = /** @class */ (function (_super) {
    tslib_1.__extends(ThemePicker, _super);
    function ThemePicker() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = {
            open: false,
            activeTabIndex: 0
        };
        return _this;
    }
    ThemePicker.prototype.componentDidUpdate = function () {
        window.requestAnimationFrame(function () {
            return window.dispatchEvent(new Event('resize'));
        });
    };
    ThemePicker.prototype.render = function () {
        var _this = this;
        var _a = this.props, selectedThemeName = _a.selectedThemeName, onThemeClick = _a.onThemeClick;
        var selectedTheme = exports.getTheme(selectedThemeName);
        return (react_1.default.createElement(menu_1.MenuSurfaceAnchor, null,
            react_1.default.createElement(menu_1.MenuSurface, { renderToPortal: true, style: { maxWidth: '100vw', width: '520px' }, open: this.state.open, onClose: function () {
                    _this.setState({ open: false });
                } },
                react_1.default.createElement(list_1.ListGroupSubheader, null, "Themes"),
                Object.keys(THEMES).map(function (themeName) {
                    var theme = exports.getTheme(themeName);
                    return (react_1.default.createElement(list_1.ListItem, { style: { cursor: 'pointer' }, key: themeName, role: "menuitem", tabIndex: 0, onClick: function (evt) {
                            evt.stopPropagation();
                            onThemeClick(themeName);
                        } },
                        react_1.default.createElement("div", { style: {
                                width: '100%',
                                height: '100%',
                                position: 'absolute',
                                top: '0',
                                left: '0'
                            } }),
                        react_1.default.createElement(list_1.ListItemGraphic, { icon: themeName === selectedThemeName ? 'check' : '' }),
                        themeName,
                        react_1.default.createElement(list_1.ListItemMeta, null,
                            react_1.default.createElement(ColorBlock, { color: theme['--mdc-theme-primary'] }),
                            react_1.default.createElement(ColorBlock, { color: theme['--mdc-theme-secondary'] }),
                            react_1.default.createElement(ColorBlock, { color: theme['--mdc-theme-background'] }),
                            react_1.default.createElement(ColorBlock, { color: theme['--mdc-theme-surface'] }))));
                }),
                react_1.default.createElement(tabs_1.TabBar, { onClick: function (evt) { return _this.setState({ open: true }); }, style: { margin: '1rem auto -1rem auto' }, activeTabIndex: this.state.activeTabIndex, onActivate: function (evt) {
                        return _this.setState({ activeTabIndex: evt.detail.index });
                    } },
                    react_1.default.createElement(tabs_1.Tab, null, "ThemeProvider"),
                    react_1.default.createElement(tabs_1.Tab, null, "CSS")),
                react_1.default.createElement(list_1.ListItem, { onClick: function () { return _this.setState({ open: true }); }, ripple: false, style: {
                        backgroundColor: 'rgba(0,0,0,.05)',
                        padding: '1rem',
                        marginTop: '1rem',
                        display: 'block',
                        height: 'auto',
                        userSelect: 'initial',
                        cursor: 'text'
                    } }, this.state.activeTabIndex === 0 ? (react_1.default.createElement("div", null,
                    react_1.default.createElement("div", { style: { whiteSpace: 'normal' } },
                        react_1.default.createElement("b", null, "Theme your App!"),
                        react_1.default.createElement("br", null),
                        "Place this tag around the root of your App, or anywhere you want to apply a custom theme.",
                        react_1.default.createElement("br", null),
                        react_1.default.createElement("br", null)),
                    react_1.default.createElement("span", { className: "token keyword" }, "import"),
                    " ",
                    '{',
                    ' ',
                    "ThemeProvider ",
                    '}',
                    " ",
                    react_1.default.createElement("span", { className: "token keyword" }, "from"),
                    ' ',
                    react_1.default.createElement("span", { className: "token string" }, "'rmwc/Theme';"),
                    react_1.default.createElement("br", null),
                    react_1.default.createElement("br", null),
                    react_1.default.createElement("span", { className: "token punctuation" }, "<"),
                    react_1.default.createElement("span", { className: "token tag" }, "ThemeProvider "),
                    react_1.default.createElement("span", { className: "token attr-name" }, "options"),
                    "=",
                    '{{',
                    Object.entries(selectedTheme).map(function (_a, index, arr) {
                        var _b = tslib_1.__read(_a, 2), t = _b[0], val = _b[1];
                        return (react_1.default.createElement("div", { key: t, style: {
                                display: 'flex',
                                justifyContent: 'space-between',
                                alignItems: 'center'
                            } },
                            react_1.default.createElement("span", null,
                                react_1.default.createElement("span", { style: { color: '#07a' } },
                                    "\u00A0\u00A0",
                                    base_1.toCamel(t.split('--mdc-theme-')[1]),
                                    ":"),
                                ' ',
                                "'",
                                val,
                                "'",
                                index < arr.length - 1 ? ',' : ''),
                            react_1.default.createElement(ColorBlock, { color: val, size: 1 })));
                    }),
                    '}}',
                    react_1.default.createElement("span", { className: "token punctuation" }, ">"),
                    react_1.default.createElement("br", null),
                    "\u00A0\u00A0",
                    react_1.default.createElement("span", { className: "token punctuation" }, "<"),
                    react_1.default.createElement("span", { className: "token tag" }, "App "),
                    react_1.default.createElement("span", { className: "token punctuation" }, "/>"),
                    react_1.default.createElement("br", null),
                    react_1.default.createElement("span", { className: "token punctuation" }, "</"),
                    react_1.default.createElement("span", { className: "token tag" }, "ThemeProvider "),
                    react_1.default.createElement("span", { className: "token punctuation" }, ">"))) : (react_1.default.createElement("div", null,
                    react_1.default.createElement("div", { style: { whiteSpace: 'normal' } },
                        react_1.default.createElement("b", null, "Theme your App!"),
                        react_1.default.createElement("br", null),
                        "Copy and paste these rules into your main css file, or a style tag in your app and customize to your liking.",
                        react_1.default.createElement("br", null),
                        react_1.default.createElement("br", null)),
                    react_1.default.createElement("span", { style: { color: '#690' } }, ":root"),
                    " ",
                    '{',
                    Object.entries(selectedTheme).map(function (_a) {
                        var _b = tslib_1.__read(_a, 2), t = _b[0], val = _b[1];
                        return (react_1.default.createElement("div", { key: t, style: {
                                display: 'flex',
                                justifyContent: 'space-between',
                                alignItems: 'center'
                            } },
                            react_1.default.createElement("span", null,
                                react_1.default.createElement("span", { style: { color: '#07a' } },
                                    "\u00A0\u00A0",
                                    t,
                                    ":"),
                                ' ',
                                val,
                                ";"),
                            react_1.default.createElement(ColorBlock, { color: val, size: 1 })));
                    }),
                    '}'))),
                react_1.default.createElement("div", { style: { padding: '1rem' } },
                    react_1.default.createElement(button_1.Button, null, "Done"))),
            react_1.default.createElement(top_app_bar_1.TopAppBarActionItem, { onClick: function () { return _this.setState({ open: !_this.state.open }); }, theme: "onPrimary", icon: "color_lens" })));
    };
    return ThemePicker;
}(react_1.default.Component));
exports.ThemePicker = ThemePicker;
exports.getTheme = function (themeName) {
    var theme = tslib_1.__assign(tslib_1.__assign({}, DEFAULT_THEME), (THEMES[themeName] || {}));
    var colors = utils_1.getAutoColorsForTheme(theme);
    var merged = tslib_1.__assign(tslib_1.__assign({}, TEXT_DEFAULTS), colors);
    var order = [
        'primary',
        'secondary',
        'error',
        'background',
        'surface',
        'onPrimary',
        'onSecondary',
        'onSurface',
        'onError',
        'textPrimaryOnBackground',
        'textSecondaryOnBackground',
        'textHintOnBackground',
        'textDisabledOnBackground',
        'textIconOnBackground',
        'textPrimaryOnLight',
        'textSecondaryOnLight',
        'textHintOnLight',
        'textDisabledOnLight',
        'textIconOnLight',
        'textPrimaryOnDark',
        'textSecondaryOnDark',
        'textHintOnDark',
        'textDisabledOnDark',
        'textIconOnDark'
    ];
    return order.reduce(function (acc, key) {
        var newKey = "--mdc-theme-" + base_1.toDashCase(key);
        acc[newKey] = merged[newKey];
        return acc;
    }, {});
};
var ColorBlock = function (_a) {
    var color = _a.color, _b = _a.size, size = _b === void 0 ? 1.5 : _b;
    return (react_1.default.createElement("div", { style: {
            display: 'inline-block',
            backgroundColor: color,
            border: '1px solid rgba(0,0,0,.25)',
            verticalAlign: 'middle',
            marginLeft: '0.5rem',
            height: size + "rem",
            width: size + "rem",
            borderRadius: '3px',
            boxSizing: 'border-box'
        } }));
};
